solutions/fully-configurable/main.tf

```diff
@@ -0,0 +1,212 @@
+  key_ring_name       = try("${local.prefix}-${var.kms_key_ring_name}", var.kms_key_ring_name)
+  key_name            = try("${local.prefix}-${var.kms_key_name}", var.kms_key_name)
+  kms_key_crn         = local.kms_encryption_enabled ? var.existing_kms_key_crn != null ? var.existing_kms_key_crn : module.kms[0].keys[format("%s.%s", local.key_ring_name, local.key_name)].crn : null
+}
+
+module "buckets" {
+  count   = local.create_buckets ? 1 : 0
+  source  = "terraform-ibm-modules/cos/ibm//modules/buckets"
+  version = "8.19.3"
+  bucket_configs = [
+    {
+      bucket_name                         = local.logs_bucket_name
+      kms_key_crn                         = local.kms_encryption_enabled ? local.kms_key_crn : null
+      kms_guid                            = local.kms_encryption_enabled ? module.existing_kms_crn_parser[0].service_instance : null
+      kms_encryption_enabled              = local.kms_encryption_enabled
+      region_location                     = var.region
+      resource_instance_id                = var.existing_cos_instance_crn
+      management_endpoint_type_for_bucket = var.management_endpoint_type_for_bucket
+    },
+    {
+      bucket_name                         = local.metrics_bucket_name
+      kms_key_crn                         = local.kms_encryption_enabled ? local.kms_key_crn : null
+      kms_guid                            = local.kms_encryption_enabled ? module.existing_kms_crn_parser[0].service_instance : null
+      kms_encryption_enabled              = local.kms_encryption_enabled
+      region_location                     = var.region
+      resource_instance_id                = var.existing_cos_instance_crn
+      management_endpoint_type_for_bucket = var.management_endpoint_type_for_bucket
+      skip_iam_authorization_policy       = true
+    }
+  ]
+}
+
+data "ibm_cos_bucket" "existing_logs_bucket" {
+  count                = local.create_buckets ? 0 : 1
+  bucket_name          = var.existing_logs_cos_bucket_name
+  resource_instance_id = var.existing_cos_instance_crn
+  bucket_region        = var.existing_logs_cos_bucket_region
+  bucket_type          = var.existing_logs_cos_bucket_type
+}
+
+data "ibm_cos_bucket" "existing_metrics_bucket" {
+  count                = local.create_buckets ? 0 : 1
+  bucket_name          = var.existing_metrics_cos_bucket_name
+  resource_instance_id = var.existing_cos_instance_crn
+  bucket_region        = var.existing_metrics_cos_bucket_region
+  bucket_type          = var.existing_metrics_cos_bucket_type
+}
+
+module "existing_kms_crn_parser" {
+  count   = var.existing_kms_instance_crn != null ? 1 : 0
+  source  = "terraform-ibm-modules/common-utilities/ibm//modules/crn-parser"
+  version = "1.1.0"
+  crn     = var.existing_kms_instance_crn
+}
+
+module "existing_kms_key_crn_parser" {
+  count   = local.use_kms_module ? 1 : 0
+  source  = "terraform-ibm-modules/common-utilities/ibm//modules/crn-parser"
+  version = "1.1.0"
+  crn     = local.kms_key_crn
+}
+
```

Comment:
You are missing support for creating KMS key and auth policy for a KMS instance in a different account using input `ibmcloud_kms_api_key`. Please be consistent with other DAs. For reference see https://github.com/terraform-ibm-modules/terraform-ibm-scc/tree/main/solutions/fully-configurable

Reply:
No this is not something we have supported anywhere else - We only support cross account KMS currently

Reply:
Hmm, although it is possible though that Event Notifications could be in a different region to the Cloud Logs instance. So for that use case, we actually added an alias for the Event Notifications instance in the SCC DA here -> https://github.com/terraform-ibm-modules/terraform-ibm-scc/blob/7713daec73b967316e998bbd6faceb59de967947/solutions/fully-configurable/provider.tf#L25-L30

However since this DA is supporting multiple EN integrations, we can't define providers dynamically. so this is going to be an issue. Can you also raise this in the deep dive today? And we should do some testing on this